README
NAME
    App::perlzonji - A more knowledgeable perldoc

VERSION
    version 1.101610

SYNOPSIS
        # perlzonji UNIVERSAL::isa
        # (runs `perldoc perlobj`)

DESCRIPTION
    "perlzonji" is like "perldoc" except it knows about more things. Try
    these:

        perlzonji xor
        perlzonji foreach
        perlzonji isa
        perlzonji AUTOLOAD
        perlzonji TIEARRAY
        perlzonji INPUT_RECORD_SEPARATOR
        perlzonji '$^F'
        perlzonji PERL5OPT
        perlzonji :mmap
        perlzonji __WARN__
        perlzonji __PACKAGE__
        perlzonji head4

    For efficiency, "alias pod=perlzonji".

    The word "zonji" means "knowledge of" in Japanese. Another example is
    the question "gozonji desu ka", meaning "Do you know?" - "go" is a
    prefix added for politeness.

FUNCTIONS
  run
    The main function, which is called by the "perlzonji" program.

  try_module
    Takes as argument the name of a module, tries to load that module and
    executes the formatter, giving that module as an argument. If loading
    the module fails, this subroutine does nothing.

  execute
    Executes the given command using "exec()". In debug mode, it also prints
    the command before executing it.

OPTIONS
    Options can be shortened according to "Case and abbreviations" in
    Getopt::Long.

    "--perldoc-command", "-c"
        Specifies the POD formatter/pager to delegate to. Default is
        "perldoc". "annopod" from AnnoCPAN::Perldoc is a better alternative.

    "--debug"
        Prints the whole command before executing it.

    "--dry-run", "-n"
        Just print the command that would be executed; don't actually
        execute it.

    "--help", "-h", "-?"
        Prints a brief help message and exits.

    "--man", "-m"
        Prints the manual page and exits.

INSTALLATION
    See perlmodinstall for information and options on installing Perl
    modules.

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests through the web interface at
    <http://rt.cpan.org>.

AVAILABILITY
    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see <http://search.cpan.org/dist/App-perlzonji/>.

    The development version lives at
    <http://github.com/hanekomu/App-perlzonji/>. Instead of sending patches,
    please fork this project using the standard git and github
    infrastructure.

AUTHOR
      Marcel Gruenauer <marcel@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2010 by Marcel Gruenauer.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Perl extension App-perlzonji

1.111500  2011-05-30 14:06:18 Europe/Vienna
    - Add UNITCHECK - and also move BEGIN CHECK INIT END to perlmod as docs are
      clearer there (thanks ranguard)

1.111470  2011-05-27 18:14:50 Europe/Vienna
    - extract find_matches() so it can be accessed directly (thanks
      ranguard)

1.101610  2010-06-10 21:50:18 Europe/Vienna
    - applied perltidy
    - added dry-run tests

1.101420  2010-05-22 22:02:43 Europe/Vienna
    - make --man and --help work again now that the docs are in the module

1.101000  2010-04-10 17:23:20 Europe/Vienna
    - added many variables from perlvar
    - made the application pluggable with Module::Pluggable and Class::Trigger

1.100770  2010-03-18 21:52:17 Europe/Vienna
    - removed weaver.ini since that's handled in Dist::Zilla's [@MARCEL] now
    - exposed %found_in as a package global
    - removed Class::MethodMaker definitions
    - added Moose definitions
    - removed duplicates
    - added dedup test

1.100710  2010-03-12 00:48:03 Europe/Vienna
    - converted the distribution to Dist::Zilla-style
    - moved the code to App::perlzonji; bin/perlzonji is just a thin wrapper
    - documented the subroutines

0.04  Thu 2010.02.18 20:59:44 CET (Marcel Gruenauer <marcel@cpan.org>)
     - added environment variables (perlrun) and PerlIO layer names
     - explained the word "zonji"

0.03  Fri 2010.02.12 11:36:44 CET (Marcel Gruenauer <marcel@cpan.org>)
     - added --perldoc-command, --debug and other common options (daxim)

0.02  Mon 2010.02.08 14:54:26 CET (Marcel Gruenauer <marcel@cpan.org>)
     - added docs to bin/perlzonji

0.01  Mon 2010.02.08 13:24:15 UTC (Marcel Gruenauer <marcel@cpan.org>)
     - original version


Makefile.PL

use strict;
use warnings;

BEGIN { require 5.008; }

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  'ABSTRACT' => 'A more knowledgeable perldoc',
  'AUTHOR' => 'Marcel Gruenauer <marcel@cpan.org>, Leo Lapworth <LLAP@cuckoo.org>',
  'BUILD_REQUIRES' => {
    'Capture::Tiny' => '0',
    'Carp' => '0',
    'English' => '0',
    'File::Find' => '0',
    'File::Temp' => '0',
    'Scalar::Util' => '0',
    'Test::More' => '0.94'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.30'
  },
  'DISTNAME' => 'App-perlzonji',
  'EXE_FILES' => [
    'bin/perlzonji'
  ],
  'LICENSE' => 'perl',
  'NAME' => 'App::perlzonji',
  'PREREQ_PM' => {
    'Class::Trigger' => '0',
    'Getopt::Long' => '0',
    'Module::Pluggable' => '0',
    'Pod::Usage' => '0'
  },
  'VERSION' => '1.111500',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);





lib/App/perlzonji/Plugin/Label.pm
use 5.008;
use strict;
use warnings;

package App::perlzonji::Plugin::Label;
BEGIN {
  $App::perlzonji::Plugin::Label::VERSION = '1.111500';
}

# ABSTRACT: Plugin for labels
use App::perlzonji;
App::perlzonji->add_trigger(
    'matches.add' => sub {
        my ($class, $word, $matches) = @_;

     # Is it a label (ends with ':')? Do this after %found_in, because there are
     # special labels such as 'SKIP:' and 'TODO:' that map to Test::More
        if ($word =~ /^\w+:$/) { push @$matches, 'perlsyn' }
    }
);
1;


__END__
=pod

=head1 NAME

App::perlzonji::Plugin::Label - Plugin for labels

=head1 VERSION

version 1.111500

=head1 SYNOPSIS

    # perlzonji FOO:
    # (runs `perldoc perlsyn`)

=head1 DESCRIPTION

This plugin for L<App::perlzonji> checks whether the search term looks like a
label and if so, adds C<perlsyn> to the match results.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org/Public/Dist/Display.html?Name=App-perlzonji>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see L<http://search.cpan.org/dist/App-perlzonji/>.

The development version lives at L<http://github.com/hanekomu/App-perlzonji>
and may be cloned from L<git://github.com/hanekomu/App-perlzonji.git>.
Instead of sending patches, please fork this project using the standard
git and github infrastructure.

=head1 AUTHORS

=over 4

=item *

Marcel Gruenauer <marcel@cpan.org>

=item *

Leo Lapworth <LLAP@cuckoo.org>

=back

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2010 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



lib/App/perlzonji/Plugin/FoundIn.pm
use 5.008;
use strict;
use warnings;

package App::perlzonji::Plugin::FoundIn;
BEGIN {
  $App::perlzonji::Plugin::FoundIn::VERSION = '1.111500';
}

# ABSTRACT: Plugin to find documentation for syntax and concepts
use App::perlzonji;

# Specify like this because it's easier. We compute the reverse later (i.e.,
# it should be easier on the hacker than on the computer).
#
# Note: 'for' is a keyword for perlpod as well ('=for'), but is listed for
# perlsyn here, as that's more likely to be the intended meaning.
our %found_in = (
    perlop => [
        qw(lt gt le ge eq ne cmp not and or xor s m tr y
          q qq qr qx qw)
    ],
    perlsyn => [qw(if else elsif unless while until for foreach)],
    perlobj => [qw(isa ISA can VERSION)],
    perlsub => [qw(AUTOLOAD DESTROY)],
    perlmod => [qw(BEGIN UNITCHECK CHECK INIT END)],
    perltie => [
        qw(TIESCALAR TIEARRAY TIEHASH TIEHANDLE FETCH STORE UNTIE
          FETCHSIZE STORESIZE POP PUSH SHIFT UNSHIFT SPLICE DELETE EXISTS
          EXTEND CLEAR FIRSTKEY NEXTKEY WRITE PRINT PRINTF READ READLINE GETC
          CLOSE)
    ],
    perlvar => [
        qw(_ a b 0 1 2 3 4 5 6 7 8 9 ARG STDIN STDOUT STDERR ARGV ENV PREMATCH
          MATCH POSTMATCH LAST_PAREN_MATCH LAST_SUBMATCH_RESULT
          LAST_MATCH_END MULTILINE_MATCHING INPUT_LINE_NUMBER NR
          INPUT_RECORD_SEPARATOR RS OUTPUT_AUTOFLUSH OUTPUT_FIELD_SEPARATOR
          OFS OUTPUT_RECORD_SEPARATOR ORS LIST_SEPARATOR SUBSCRIPT_SEPARATOR
          SUBSEP FORMAT_PAGE_NUMBER FORMAT_LINES_PER_PAGE FORMAT_LINES_LEFT
          LAST_MATCH_START FORMAT_NAME FORMAT_TOP_NAME
          FORMAT_LINE_BREAK_CHARACTERS FORMAT_FORMFEED ACCUMULATOR
          CHILD_ERROR CHILD_ERROR_NATIVE ENCODING OS_ERROR ERRNO
          EXTENDED_OS_ERROR EVAL_ERROR PROCESS_ID PID REAL_USER_ID UID
          EFFECTIVE_USER_ID EUID REAL_GROUP_ID GID EFFECTIVE_GROUP_ID EGID
          PROGRAM_NAME COMPILING DEBUGGING RE_DEBUG_FLAGS RE_TRIE_MAXBUF
          SYSTEM_FD_MAX INPLACE_EDIT OSNAME OPEN PERLDB
          LAST_REGEXP_CODE_RESULT EXCEPTIONS_BEING_CAUGHT BASETIME TAINT
          UNICODE UTF8CACHE UTF8LOCALE PERL_VERSION WARNING WARNING_BITS
          WIN32_SLOPPY_STAT EXECUTABLE_NAME ARGVOUT INC SIG __DIE__ __WARN__
          $& $` $' $+ $^N @+ %+ $. $/ $| $\ $" $; $% $= $- @- %- $~ $^ $:
          $? $! %! $@ $$ $< $> $[ $] $^A $^C $^D $^E $^F $^H $^I $^L
          $^M $^O $^P $^R $^S $^T $^V $^W $^X %^H @F @_
          ), '$,', '$(', '$)',
    ],
    perlrun => [
        qw(HOME LOGDIR PATH PERL5LIB PERL5OPT PERLIO PERLIO_DEBUG PERLLIB
          PERL5DB PERL5DB_THREADED PERL5SHELL PERL_ALLOW_NON_IFS_LSP
          PERL_DEBUG_MSTATS PERL_DESTRUCT_LEVEL PERL_DL_NONLAZY PERL_ENCODING
          PERL_HASH_SEED PERL_HASH_SEED_DEBUG PERL_ROOT PERL_SIGNALS
          PERL_UNICODE)
    ],
    perlpod => [
        qw(head1 head2 head3 head4 over item back cut pod begin
          end)
    ],
    perldata => [qw(__FILE__ __LINE__ __PACKAGE__)],

    # We could also list common functions and methods provided by some
    # commonly used modules, like:
    Moose => [
        qw(has before after around super override inner augment confessed
          extends with)
    ],
    Error        => [qw(try catch except otherwise finally record)],
    SelfLoader   => [qw(__DATA__ __END__ DATA)],
    Storable     => [qw(freeze thaw)],
    Carp         => [qw(carp cluck croak confess shortmess longmess)],
    'Test::More' => [
        qw(plan use_ok require_ok ok is isnt like unlike cmp_ok
          is_deeply diag can_ok isa_ok pass fail eq_array eq_hash eq_set skip
          todo_skip builder SKIP: TODO:)
    ],
    'Getopt::Long' => [qw(GetOptions)],
    'File::Find'   => [qw(find finddepth)],
    'File::Path'   => [qw(mkpath rmtree)],
    'File::Spec'   => [
        qw(canonpath catdir catfile curdir devnull rootdir
          tmpdir updir no_upwards case_tolerant file_name_is_absolute path
          splitpath splitdir catpath abs2rel rel2abs)
    ],
    'File::Basename' => [
        qw(fileparse fileparse_set_fstype basename
          dirname)
    ],
    'File::Temp' => [
        qw(tempfile tempdir tmpnam tmpfile mkstemp mkstemps
          mkdtemp mktemp unlink0 safe_level)
    ],
    'File::Copy' => [qw(copy move cp mv rmscopy)],
    'PerlIO' =>
      [qw(:bytes :crlf :mmap :perlio :pop :raw :stdio :unix :utf8 :win32)],
);
App::perlzonji->add_trigger(
    'matches.add' => sub {
        my ($class, $word, $matches) = @_;
        while (my ($file, $words) = each our %found_in) {
            for (@$words) {
                push @$matches, $file if $_ eq $word;
            }
        }
    }
);
1;


__END__
=pod

=head1 NAME

App::perlzonji::Plugin::FoundIn - Plugin to find documentation for syntax and concepts

=head1 VERSION

version 1.111500

=head1 SYNOPSIS

    # perlzonji elsif
    # (runs `perldoc perlsyn`)

=head1 DESCRIPTION

This plugin for L<App::perlzonji> knows where to find documentation for syntax
and built-in Perl concepts. It knows about things like

    elsif
    VERSION
    END
    OUTPUT_RECORD_SEPARATOR
    $^R
    PERL5OPT
    head3
    __DATA__
    :utf8

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org/Public/Dist/Display.html?Name=App-perlzonji>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see L<http://search.cpan.org/dist/App-perlzonji/>.

The development version lives at L<http://github.com/hanekomu/App-perlzonji>
and may be cloned from L<git://github.com/hanekomu/App-perlzonji.git>.
Instead of sending patches, please fork this project using the standard
git and github infrastructure.

=head1 AUTHORS

=over 4

=item *

Marcel Gruenauer <marcel@cpan.org>

=item *

Leo Lapworth <LLAP@cuckoo.org>

=back

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2010 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



lib/App/perlzonji/Plugin/UseModule.pm
use 5.008;
use strict;
use warnings;

package App::perlzonji::Plugin::UseModule;
BEGIN {
  $App::perlzonji::Plugin::UseModule::VERSION = '1.111500';
}

# ABSTRACT: Plugin to try the search word as a module name
use App::perlzonji;
App::perlzonji->add_trigger(
    'matches.add' => sub {
        my ($class, $word, $matches) = @_;
        my $try_module = sub {
            my $module = shift;
            eval "use $module;";
            !$@ && push @$matches, $module;
        };

        # try it as a module
        $try_module->($word);

     # if it contains '::', it's not a function - strip off the last bit and try
     # that again as a module
        $word =~ s/::(\w+)$// && $try_module->($word);
    }
);
1;
1;


__END__
=pod

=head1 NAME

App::perlzonji::Plugin::UseModule - Plugin to try the search word as a module name

=head1 VERSION

version 1.111500

=head1 SYNOPSIS

    # perlzonji Devel::SearchINC

=head1 DESCRIPTION

This plugin for L<App::perlzonji> tries to use the search term as a module
name. If the module can be loaded, it is added to the match results. In case
the search term looks like a fully qualified function name such as
C<Foo::Bar::some_function>, the function part is stripped off the the
remainder is tried again.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org/Public/Dist/Display.html?Name=App-perlzonji>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see L<http://search.cpan.org/dist/App-perlzonji/>.

The development version lives at L<http://github.com/hanekomu/App-perlzonji>
and may be cloned from L<git://github.com/hanekomu/App-perlzonji.git>.
Instead of sending patches, please fork this project using the standard
git and github infrastructure.

=head1 AUTHORS

=over 4

=item *

Marcel Gruenauer <marcel@cpan.org>

=item *

Leo Lapworth <LLAP@cuckoo.org>

=back

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2010 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



lib/App/perlzonji/Plugin/SpecialPackages.pm
use 5.008;
use strict;
use warnings;

package App::perlzonji::Plugin::SpecialPackages;
BEGIN {
  $App::perlzonji::Plugin::SpecialPackages::VERSION = '1.111500';
}

# ABSTRACT: Plugin to find documentation for special Perl packages
use App::perlzonji;
App::perlzonji->add_trigger(
    'matches.add' => sub {
        my ($class, $word, $matches) = @_;
        $word =~ /^UNIVERSAL::/ && push @$matches, 'perlobj';
        $word =~ /^CORE::/      && push @$matches, 'perlsub';
    }
);
1;


__END__
=pod

=head1 NAME

App::perlzonji::Plugin::SpecialPackages - Plugin to find documentation for special Perl packages

=head1 VERSION

version 1.111500

=head1 SYNOPSIS

    # perlzonji UNIVERSAL::isa
    # (runs `perldoc perlobj`)

=head1 DESCRIPTION

This plugin for L<App::perlzonji> knows where special Perl packages like
C<UNIVERSAL::> and C<CORE::> are documented.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org/Public/Dist/Display.html?Name=App-perlzonji>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see L<http://search.cpan.org/dist/App-perlzonji/>.

The development version lives at L<http://github.com/hanekomu/App-perlzonji>
and may be cloned from L<git://github.com/hanekomu/App-perlzonji.git>.
Instead of sending patches, please fork this project using the standard
git and github infrastructure.

=head1 AUTHORS

=over 4

=item *

Marcel Gruenauer <marcel@cpan.org>

=item *

Leo Lapworth <LLAP@cuckoo.org>

=back

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2010 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



lib/App/perlzonji.pm
use 5.008;
use strict;
use warnings;

package App::perlzonji;
BEGIN {
  $App::perlzonji::VERSION = '1.111500';
}

# ABSTRACT: A more knowledgeable perldoc
use Getopt::Long;
use Pod::Usage;
use Class::Trigger;
use Module::Pluggable require => 1;
__PACKAGE__->plugins;    # 'require' them

sub run {
    our %opt = ('perldoc-command' => 'perldoc');
    GetOptions(\%opt, qw(help|h|? man|m perldoc-command|c:s debug dry-run|n))
      or pod2usage(2);
    if ($opt{help}) {
        pod2usage(
            -exitstatus => 0,
            -input      => __FILE__,
        );
    }
    if ($opt{man}) {
        pod2usage(
            -exitstatus => 0,
            -input      => __FILE__,
            -verbose    => 2
        );
    }
    my $word    = shift @ARGV;
    my $matches = find_matches($word);
    if (@$matches) {
        if (@$matches > 1) {
            warn sprintf "%s matches for [%s], using first (%s):\n",
              scalar(@$matches), $word, $matches->[0];
            warn "    $_\n" for @$matches;
        }
        execute($opt{'perldoc-command'}, $matches->[0]);
    }

    # fallback
    warn "assuming that [$word] is a built-in function\n" if $opt{debug};
    execute($opt{'perldoc-command'}, qw(-f), $word);
}

sub find_matches {
    my $word = shift;
    my @matches;
    __PACKAGE__->call_trigger('matches.add', $word, \@matches);
    return \@matches;
}

sub execute {
    our %opt;
    if ($opt{'dry-run'}) {

        # under --dry-run, don't pretend to execute more than once
        our $executed;
        return if $executed++;
        warn "@_\n";
    } else {
        warn "@_\n" if $opt{debug};
        exec @_;
    }
}
1;


__END__
=pod

=for stopwords Dieckow gozonji desu ka

=head1 NAME

App::perlzonji - A more knowledgeable perldoc

=head1 VERSION

version 1.111500

=head1 SYNOPSIS

    # perlzonji UNIVERSAL::isa
    # (runs `perldoc perlobj`)

=head1 DESCRIPTION

C<perlzonji> is like C<perldoc> except it knows about more things. Try these:

    perlzonji xor
    perlzonji foreach
    perlzonji isa
    perlzonji AUTOLOAD
    perlzonji TIEARRAY
    perlzonji INPUT_RECORD_SEPARATOR
    perlzonji '$^F'
    perlzonji PERL5OPT
    perlzonji :mmap
    perlzonji __WARN__
    perlzonji __PACKAGE__
    perlzonji head4

For efficiency, C<alias pod=perlzonji>.

The word C<zonji> means "knowledge of" in Japanese. Another example
is the question "gozonji desu ka", meaning "Do you know?" - "go" is a
prefix added for politeness.

=head1 FUNCTIONS

=head2 run

The main function, which is called by the C<perlzonji> program.

=head2 try_module

Takes as argument the name of a module, tries to load that module and
executes the formatter, giving that module as an argument. If loading
the module fails, this subroutine does nothing.

=head2 execute

Executes the given command using C<exec()>. In debug mode, it also
prints the command before executing it.

=head2 find_matches

Takes a word and returns the matches for that word. It's in a separate
function to separate logic from presentation so other programs can use
this module as well.

=head1 OPTIONS

Options can be shortened according to L<Getopt::Long/"Case and
abbreviations">.

=over

=item C<--perldoc-command>, C<-c>

Specifies the POD formatter/pager to delegate to. Default is         .
C<perldoc> C<annopod> from L<AnnoCPAN::Perldoc> is a better          .
alternative                                                          .

=item C<--debug>

Prints the whole command before executing it.

=item C<--dry-run>, C<-n>

Just print the command that would be executed; don't actually execute
it.

=item C<--help>, C<-h>, C<-?>

Prints a brief help message and exits.

=item C<--man>, C<-m>

Prints the manual page and exits.

=back

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org/Public/Dist/Display.html?Name=App-perlzonji>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see L<http://search.cpan.org/dist/App-perlzonji/>.

The development version lives at L<http://github.com/hanekomu/App-perlzonji>
and may be cloned from L<git://github.com/hanekomu/App-perlzonji.git>.
Instead of sending patches, please fork this project using the standard
git and github infrastructure.

=head1 AUTHORS

=over 4

=item *

Marcel Gruenauer <marcel@cpan.org>

=item *

Leo Lapworth <LLAP@cuckoo.org>

=back

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2010 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut

